virttest/guest_agent.py

```diff
@@ -539,10 +538,10 @@ def set_user_password(self, password, crypted=False, username="root"):
 
         if crypted:
             openssl_cmd = "openssl passwd -crypt %s" % password
-            password = decode_to_text(process.system_output(openssl_cmd)).strip('\n')
+            password = process.system_output(openssl_cmd).strip('\n')
```

Comment:
That is because this line is not correct.
```python
[root@dhcp-8-192 ~]# python
Python 3.5.5 |Anaconda, Inc.| (default, May 13 2018, 21:12:35) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> password = b"redhat"
>>> password.strip("\n")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
>>> import base64
>>> a = base64.b64encode(password)
>>> a
b'cmVkaGF0'
>>> 
```

Reply:
@hereischen No, I am sure the fail one is b64encode():
```
01:53:23 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_plugins_vt-63.0-py3.6.egg/virttest/guest_agent.py", line 547, in set_user_password
01:53:23 ERROR|     "password": base64.b64encode(password).decode()}
01:53:23 ERROR|   File "/usr/lib64/python3.6/base64.py", line 58, in b64encode
01:53:23 ERROR|     encoded = binascii.b2a_base64(s, newline=False)
01:53:23 ERROR| TypeError: a bytes-like object is required, not 'str'
```
I guess the real reason is `crypted` is `False`, so the password actually not comes from `process.system_output()`, it is passed by the caller, so it is text.

Reply:
@hereischen So `decode_to_text()` should not be blamed, I will bring it back.